=== modified file 'csharp/ICT/BuildTools/GenerateSQL/writesql.cs'
--- a/csharp/ICT/BuildTools/GenerateSQL/writesql.cs	2014-03-07 17:19:37 +0000
+++ b/csharp/ICT/BuildTools/GenerateSQL/writesql.cs	2014-10-02 04:39:28 +0000
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -141,6 +141,11 @@
             FileMode.Create, FileAccess.Write);
         sw = new StreamWriter(outPutFileStream);
 
+        if (ATargetDatabase == eDatabaseType.PostgreSQL)
+        {
+            sw.WriteLine("SET client_min_messages TO WARNING;");
+        }
+
         foreach (TTable Table in Tables)
         {
             if (!WriteTable(ATargetDatabase, sw, Table))
@@ -221,6 +226,11 @@
             FileMode.Create, FileAccess.Write);
         sw = new StreamWriter(outPutFileStream);
 
+        if (ATargetDatabase == eDatabaseType.PostgreSQL)
+        {
+            sw.WriteLine("SET client_min_messages TO WARNING;");
+        }
+
         foreach (TTable Table in Tables)
         {
             if (!WriteTable(ATargetDatabase, sw, Table))

