diff --git a/csharp/ICT/Common/IO/SmtpEmail.cs b/csharp/ICT/Common/IO/SmtpEmail.cs
index 6050974..b2d9b42 100644
--- a/csharp/ICT/Common/IO/SmtpEmail.cs
+++ b/csharp/ICT/Common/IO/SmtpEmail.cs
@@ -179,8 +179,6 @@ namespace Ict.Common.IO
             }
         }
 
-        private static List <DateTime>FEmailsSentInLastMinute = new List <DateTime>();
-
         /// <summary>
         /// Send an email message
         /// </summary>
@@ -206,39 +204,32 @@ namespace Ict.Common.IO
             {
                 AEmail.IsBodyHtml = AEmail.Body.ToLower().Contains("<html>");
 
-                int LimitEmailsPerMinute = TAppSettingsManager.GetInt32("SmtpLimitMessagesPerMinute", 30);
-                int countInLastMinute = LimitEmailsPerMinute;
+                int AttemptCount = 3;
 
-                while (countInLastMinute >= LimitEmailsPerMinute)
+                while (AttemptCount > 0)
                 {
-                    countInLastMinute = 0;
-
-                    foreach (DateTime dt in FEmailsSentInLastMinute)
+                    AttemptCount--;
+                    try
                     {
-                        // better check the last 2 minutes, to avoid confusion
-                        if (DateTime.Compare(dt.AddMinutes(2), DateTime.Now) >= 0)
-                        {
-                            countInLastMinute++;
-                        }
-                    }
+                        // for office365, this takes about 15 seconds
+                        FSmtpClient.Send(AEmail);
 
-                    if (countInLastMinute == 0)
-                    {
-                        FEmailsSentInLastMinute.Clear();
-                    }
+                        AEmail.Headers.Add("Date-Sent", DateTime.Now.ToString());
 
-                    if (countInLastMinute >= LimitEmailsPerMinute)
+                        return true;
+                    }
+                    catch (Exception e)
                     {
-                        Thread.Sleep(TimeSpan.FromSeconds(10));
+                        if (AttemptCount > 0)
+                        {
+                            Thread.Sleep(TimeSpan.FromMinutes(1));
+                        }
+                        else
+                        {
+                            throw e;
+                        }
                     }
                 }
-
-                FEmailsSentInLastMinute.Add(DateTime.Now);
-
-                FSmtpClient.Send(AEmail);
-
-                AEmail.Headers.Add("Date-Sent", DateTime.Now.ToString());
-                return true;
             }
             catch (Exception ex)
             {
@@ -257,6 +248,8 @@ namespace Ict.Common.IO
 
                 throw;
             }
+
+            return false;
         }
     }
 }
\ No newline at end of file
