--- a/csharp/ICT/Petra/Client/MReporting/logic/Calculator.cs	2014-07-17 14:50:29 +0000
+++ b/csharp/ICT/Petra/Client/MReporting/logic/Calculator.cs	2014-09-29 14:32:56 +0000
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2013 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -487,6 +487,8 @@
 
                 // UnRegister Object from the TEnsureKeepAlive Class so that the Object can get GC'd on the PetraServer
                 TEnsureKeepAlive.UnRegister(FReportingGenerator);
+
+                return false;
             }
 
             // todo: allow canceling of the calculation of a report
@@ -497,6 +499,9 @@
 
             ReturnValue = FReportingGenerator.GetSuccess();
 
+            // UnRegister Object from the TEnsureKeepAlive Class so that the Object can get GC'd on the PetraServer
+            TEnsureKeepAlive.UnRegister(FReportingGenerator);
+
             if (ReturnValue)
             {
                 if (FCalculatesExtract)
@@ -551,8 +556,6 @@
                             TLogging.Log(FReportingGenerator.GetErrorMessage());
                         }
 
-                        // UnRegister Object from the TEnsureKeepAlive Class so that the Object can get GC'd on the PetraServer
-                        TEnsureKeepAlive.UnRegister(FReportingGenerator);
                         FKeepUpProgressCheck = false;
                         break;
 

