=== modified file 'csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs'
--- a/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs	2014-03-07 17:19:37 +0000
+++ b/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs	2014-09-18 14:38:14 +0000
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop, christiank
 //
-// Copyright 2004-2013 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -103,10 +103,7 @@
             {
                 TRemote.MCommon.WebConnectors.CancelJob();
 
-                this.DialogResult = DialogResult.Cancel;
-
                 FConfirmedClosing = true;
-                Close();
             }
         }
 
@@ -128,7 +125,7 @@
 
                 if (finished)
                 {
-                    this.DialogResult = DialogResult.OK;
+                    this.DialogResult = FCancelled?DialogResult.Cancel:DialogResult.OK;
                     FConfirmedClosing = true;
                     Close();
                 }

=== modified file 'csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs'
--- a/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs	2014-09-18 14:38:14 +0000
+++ b/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs	2014-09-18 15:09:54 +0000
@@ -37,6 +37,7 @@
     {
         private bool FConfirmedClosing = false;
         private bool FShowCancellationConfirmationQuestion = false;
+        private bool FCancelled = false;
 
         /// <summary>
         /// Constructor
@@ -102,8 +103,7 @@
             if (CancelConfirmationResult == DialogResult.Yes)
             {
                 TRemote.MCommon.WebConnectors.CancelJob();
-
-                FConfirmedClosing = true;
+                FCancelled = true;
             }
         }
 
@@ -125,7 +125,7 @@
 
                 if (finished)
                 {
-                    this.DialogResult = FCancelled?DialogResult.Cancel:DialogResult.OK;
+                    this.DialogResult = FCancelled ? DialogResult.Cancel : DialogResult.OK;
                     FConfirmedClosing = true;
                     Close();
                 }

=== modified file 'csharp/ICT/Petra/Server/lib/MConference/web/DeleteConference.cs'
--- a/csharp/ICT/Petra/Server/lib/MConference/web/DeleteConference.cs	2014-03-07 17:19:37 +0000
+++ b/csharp/ICT/Petra/Server/lib/MConference/web/DeleteConference.cs	2014-09-18 15:09:54 +0000
@@ -96,6 +96,7 @@
 
                 if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                 {
+                    TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                     throw new Exception("Deletion of Conference was cancelled by the user");
                 }
 

=== modified file 'csharp/ICT/Petra/Server/lib/MFinance/BankImport/Matching.cs'
--- a/csharp/ICT/Petra/Server/lib/MFinance/BankImport/Matching.cs	2014-03-07 17:19:37 +0000
+++ b/csharp/ICT/Petra/Server/lib/MFinance/BankImport/Matching.cs	2014-09-18 15:09:54 +0000
@@ -70,6 +70,7 @@
 
                 if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                 {
+                    TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                     return;
                 }
 

=== modified file 'csharp/ICT/Petra/Server/lib/MFinance/setup/GL.Setup.cs'
--- a/csharp/ICT/Petra/Server/lib/MFinance/setup/GL.Setup.cs	2014-09-02 15:24:23 +0000
+++ b/csharp/ICT/Petra/Server/lib/MFinance/setup/GL.Setup.cs	2014-09-18 15:09:54 +0000
@@ -3013,6 +3013,7 @@
 
                 if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                 {
+                    TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                     throw new Exception("Deletion of Ledger was cancelled by the user");
                 }
 

=== modified file 'csharp/ICT/Petra/Server/lib/MPartner/web/MergePartners.cs'
--- a/csharp/ICT/Petra/Server/lib/MPartner/web/MergePartners.cs	2014-07-17 14:50:29 +0000
+++ b/csharp/ICT/Petra/Server/lib/MPartner/web/MergePartners.cs	2014-09-18 15:09:54 +0000
@@ -428,6 +428,7 @@
 
                 if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                 {
+                    TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                     DBAccess.GDBAccessObj.RollbackTransaction();
                     throw new Exception("Partner merge was cancelled by the user");
                 }

=== modified file 'csharp/ICT/Petra/Server/lib/MSysMan/ImportExport.cs'
--- a/csharp/ICT/Petra/Server/lib/MSysMan/ImportExport.cs	2014-03-07 17:19:37 +0000
+++ b/csharp/ICT/Petra/Server/lib/MSysMan/ImportExport.cs	2014-09-18 15:09:54 +0000
@@ -260,6 +260,7 @@
 
                 if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                 {
+                    TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                     DBAccess.GDBAccessObj.RollbackTransaction();
                     return false;
                 }
@@ -296,6 +297,7 @@
 
                 if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                 {
+                    TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                     DBAccess.GDBAccessObj.RollbackTransaction();
                     return false;
                 }
@@ -324,6 +326,7 @@
 
                     if (TProgressTracker.GetCurrentState(DomainManager.GClientID.ToString()).CancelJob == true)
                     {
+                        TProgressTracker.FinishJob(DomainManager.GClientID.ToString());
                         DBAccess.GDBAccessObj.RollbackTransaction();
                         return false;
                     }

