--- a/csharp/ICT/Petra/Client/app/Core/UserDefaults.cs	2014-06-30 11:04:59 +0000
+++ b/csharp/ICT/Petra/Client/app/Core/UserDefaults.cs	2014-06-30 13:13:36 +0000
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank, timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -435,22 +435,22 @@
 
         /// <summary>
         /// Saves all changed User Defaults.
-        ///
         /// </summary>
-        /// <returns>true if successful, false if not.
-        /// </returns>
         public static void SaveChangedUserDefaults()
         {
             SUserDefaultsTable UserDefaultsDataTableChanges;
 
             UserDefaultsDataTableChanges = UUserDefaultsDataTable.GetChangesTyped();
 
-            // MessageBox.Show('Changed/added User Defaults: ' + UserDefaultsDataTableChanges.Rows.Count.ToString);
-            TRemote.MSysMan.Maintenance.UserDefaults.WebConnectors.SaveUserDefaults(Ict.Petra.Shared.UserInfo.GUserInfo.UserID,
-                ref UserDefaultsDataTableChanges);
+            if (UserDefaultsDataTableChanges != null)
+            {
+                // MessageBox.Show('Changed/added User Defaults: ' + UserDefaultsDataTableChanges.Rows.Count.ToString);
+                TRemote.MSysMan.Maintenance.UserDefaults.WebConnectors.SaveUserDefaults(Ict.Petra.Shared.UserInfo.GUserInfo.UserID,
+                    ref UserDefaultsDataTableChanges);
 
-            // TODO 1 oChristianK cUserDefaults / ModificationID : Copy the ModificationID into the Client's DataTable so that the PetraClient's ModificationID's of UserDefaults are the same than the ones of the PetraServer.
-            UUserDefaultsDataTable.AcceptChanges();
+                // TODO 1 oChristianK cUserDefaults / ModificationID : Copy the ModificationID into the Client's DataTable so that the PetraClient's ModificationID's of UserDefaults are the same than the ones of the PetraServer.
+                UUserDefaultsDataTable.AcceptChanges();
+            }
         }
 
         /// <summary>

